[aotinductor] Enable test_non_default_cuda_device on CI

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110510
* __->__ #110509

Summary: test_non_default_cuda_device needs to run on a multi-gpu CI instance

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Differential Revision: [D49937115](https://our.internmc.facebook.com/intern/diff/D49937115)

desertfire committed:
[aotinductor] Enable test_non_default_cuda_device on CI

Summary: test_non_default_cuda_device needs to run on a multi-gpu CI instance

[ghstack-poisoned]

khabinov reviewed (approved):
Thx!

chenyang78 commented on test/inductor/test_aot_inductor.py:
```diff
@@ -670,10 +683,11 @@ def forward(self, x, y):
+    @requires_multigpu()
     def test_non_default_cuda_device(self):
+        if self.device == "cpu":
+            raise unittest.SkipTest("requires CUDA")
```
Wondering if ```@requires_cuda()``` decorator would work or not?

desertfire replied:
That will be a problem when running on a gpu instance, where both cuda and cpu version of this test will be run.

desertfire committed:
Update on "[aotinductor] Enable test_non_default_cuda_device on CI"


Summary: test_non_default_cuda_device needs to run on a multi-gpu CI instance

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D49937115).
